src/main/java/org/opensearch/security/dlic/rest/api/SecuritySSLCertsApiAction.java

```diff
@@ -138,65 +138,70 @@ boolean accessHandler(final RestRequest request) {
-                .field("transport_certificates_list", generateCertDetailList(keyStore.getTransportCerts()))
+                .field(
+                    "http_certificates_list",
+                    generateCertDetailList(
+                        sslSettingsManager.sslContextHandler(SSL_HTTP_PREFIX).map(SslContextHandler::keyMaterialCertificates).orElse(null)
+                    )
+                )
+                .field(
+                    "transport_certificates_list",
+                    generateCertDetailList(
+                        sslSettingsManager.sslContextHandler(SSL_TRANSPORT_PREFIX)
+                            .map(SslContextHandler::keyMaterialCertificates)
+                            .orElse(null)
+                    )
+                )
                 .endObject()
         );
     }
 
-    private List<Map<String, String>> generateCertDetailList(final X509Certificate[] certs) {
+    private List<Map<String, String>> generateCertDetailList(final Stream<Certificate> certs) {
         if (certs == null) {
             return null;
         }
-        return Arrays.stream(certs).map(cert -> {
-            final String issuerDn = cert != null && cert.getIssuerX500Principal() != null ? cert.getIssuerX500Principal().getName() : "";
-            final String subjectDn = cert != null && cert.getSubjectX500Principal() != null ? cert.getSubjectX500Principal().getName() : "";
-
-            final String san = securityKeyStore.getSubjectAlternativeNames(cert);
-
-            final String notBefore = cert != null && cert.getNotBefore() != null ? cert.getNotBefore().toInstant().toString() : "";
-            final String notAfter = cert != null && cert.getNotAfter() != null ? cert.getNotAfter().toInstant().toString() : "";
-            return ImmutableMap.of(
+        return certs.map(
+            c -> ImmutableMap.of(
                 "issuer_dn",
-                issuerDn,
+                c.issuer(),
                 "subject_dn",
-                subjectDn,
+                c.subject(),
                 "san",
-                san,
+                c.subjectAlternativeNames(),
                 "not_before",
-                notBefore,
+                c.notBefore(),
                 "not_after",
-                notAfter
-            );
-        }).collect(Collectors.toList());
+                c.notAfter()
+            )
+        ).collect(Collectors.toList());
     }
 
-    protected void reloadCertificates(final RestChannel channel, final RestRequest request, final SecurityKeyStore keyStore)
-        throws IOException {
+    protected void reloadCertificates(final RestChannel channel, final RestRequest request) throws IOException {
         final String certType = request.param("certType").toLowerCase().trim();
         try {
             switch (certType) {
                 case "http":
-                    if (!httpsEnabled) {
+                    if (sslSettingsManager.sslConfiguration(SSL_HTTP_PREFIX).isPresent()) {
```

Comment:
Why not check if its present and set to false? Should `reloadSslContext` be called if HTTPS is disabled?

Reply:
Yes, we can disable HTTPs, but transport and the transport client will always require TLS.

Reply:
Doesn't this change the current logic?

Currently, if https is set to false, the API returns BAD_REQUEST with a message `SSL for HTTP is disabled`.

With these changes, won't reload always be called as long as a setting named `plugins.security.ssl.http.enabled` is present in `opensearch.yml`?

I guess you can reload the certs if https is disabled, but what's the point?

Reply:
No, the logic remains the same as before. The HTTP SSL configuration will only appear if it has been enabled in the `opensearch.yml`. 
I just removed httpsEnabled = getAsBoolean(SSLConfigConstants.SECURITY_SSL_HTTP_ENABLED, true) since we can now rely on `sslSettingsManager`.

Reply:
Ahh, I see what you mean. This is indeed a strange solution. The issue is that we can't reload the HTTP SSL configuration under any circumstances because `opensearch.yml` only contains static settings. The current solution doesn't read the HTTP settings even if they've been configured but are disabled. Here:
https://github.com/opensearch-project/security/blob/31cfd54a4166af232fb1d584720de29fc0620da0/src/main/java/org/opensearch/security/ssl/DefaultSecurityKeyStore.java#L316